sdk/core/azure-core/inc/azure/core/internal/unique_handle.hpp

```diff
@@ -3,65 +3,68 @@
 #pragma once
 #include <memory>
 
-namespace Azure { namespace Core { namespace _internal {
+namespace Azure { namespace Core {
+  namespace _detail {
```

Comment:
Should this be in Azure::Core::Http::_detail?

Reply:
It could be, does not sound like a bad idea from a point of organizing stuff.
However, an argument against - the same UniqueHelper template specialization within the same library would probably have the same implementation - so it will be good if it is actually reused - no problem with single library having all its Helper template specializations in the same namespace, it could be even convenient that all of them are there.
Plus - it is less code - we don' need to create `core_http_unique_handle.hpp`, and `core_credentials_unique_handle.hpp`, and define the very same types (`template<typename> struct UniqueHandleHelper;` in each namespace).

`UniqueHandle` is `Azure::Core::_internal`, not `Azure::Core::Http::_internal` - unlike UniqueHandleHelper, which is debateable, there is nothing Core-HTTP specific about the `UniqueHandle<T>`.
I get it, why it is per-library, but per namespace within the same library sounds a bit overkill.